File: app/src/main/java/com/compuspulse/CampusPulseApp.java
package com.compuspulse;

import android.app.Application;

import com.compuspulse.data.local.db.AppDatabase;

public class CampusPulseApp extends Application {

    private static CampusPulseApp instance;
    private AppDatabase database;

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        database = AppDatabase.getInstance(this);
    }

    public static CampusPulseApp getInstance() {
        return instance;
    }

    public AppDatabase getDatabase() {
        return database;
    }
}

File: app/src/main/java/com/compuspulse/CampusPulseApp.kt
package com.compuspulse

import android.app.Application
import com.compuspulse.data.local.db.AppDatabase

class CampusPulseApp : Application() {

    lateinit var database: AppDatabase
        private set

    override fun onCreate() {
        super.onCreate()
        instance = this
        database = AppDatabase.getInstance(this)
    }

    companion object {
        private lateinit var instance: CampusPulseApp

        @JvmStatic
        fun getInstance(): CampusPulseApp = instance
    }
}
